src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -0,0 +1,120 @@
+import { Buffer } from "buffer";
+import {
+  Paginate,
+  Cip30DataSignature,
+  IWalletInfo,
+} from "@fabianbormann/cardano-peer-connect/dist/src/types";
+import { CardanoPeerConnect } from "@fabianbormann/cardano-peer-connect";
+import { Signer } from "signify-ts";
+import { Agent } from "../../agent/agent";
+
+class IdentityWalletConnect extends CardanoPeerConnect {
+  static readonly IDENTIFIER_ID_NOT_LOCATED =
+    "The id doesn't correspond with any stored identifier";
+  static readonly NO_IDENTIFIERS_STORED = "No stored identifiers";
+  static readonly AID_MISSING_SIGNIFY_NAME =
+    "Metadata record for KERI AID is missing the Signify name";
+
+  getIdentifierOobi: () => Promise<string>;
+  sign: (oobi: string, payload: string) => Promise<string>;
+  generateOobi: (identifierId: string) => Promise<string>;
+
+  signerCache: Map<string, Signer>;
+
+  constructor(
+    walletInfo: IWalletInfo,
+    seed: string | null,
+    announce: string[],
+    discoverySeed?: string | null
+  ) {
+    super(walletInfo, {
+      seed: seed,
+      announce: announce,
+      discoverySeed: discoverySeed,
+      logLevel: "info",
+    });
+
+    this.signerCache = new Map();
+
+    this.getIdentifierOobi = async (): Promise<string> => {
```

Comment:
Hmm, we have this and `generateOobi` - don't we just need these one, but return the oobi here?